src/cargo/ops/cargo_new.rs

```diff
@@ -261,6 +262,19 @@ fn check_name(
     Ok(())
 }
 
+/// Checks if the path contains any invalid PATH env characters.
+fn check_path(path: &Path, shell: &mut Shell) -> CargoResult<()> {
+    // warn if the path contains characters that will break `env::join_paths`
+    if let Err(_) = paths::join_paths(slice::from_ref(&OsStr::new(path)), "") {
```

Comment:
What about simply calling `std::env::join_paths(path)`?

```suggestion
    if let Err(_) = std::env::join_paths(path) {
```